packages/composites/glv-token/src/transport/price.ts

```diff
@@ -0,0 +1,380 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+
+import { BaseEndpointTypes, inputParameters } from '../endpoint/price'
+import { ethers, utils } from 'ethers'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import {
+  EndpointContext,
+  LwbaResponseDataFields,
+} from '@chainlink/external-adapter-framework/adapter'
+import { AdapterResponse, makeLogger, sleep } from '@chainlink/external-adapter-framework/util'
+import {
+  toFixed,
+  median,
+  PriceData,
+  SIGNED_PRICE_DECIMALS,
+  Source,
+  Token,
+  Market,
+  mapSymbol,
+  mapParameter,
+} from './utils'
+import abi from '../config/readerAbi.json'
+import glvAbi from '../config/glvReaderAbi.json'
+import { AdapterDataProviderError } from '@chainlink/external-adapter-framework/validation/error'
+
+const logger = makeLogger('GlvToken')
+
+type RequestParams = typeof inputParameters.validated
+
+export type GlvTokenTransportTypes = BaseEndpointTypes
+
+interface glvInformation {
+  glvToken: string
+  longToken: Token
+  shortToken: Token
+  markets: Record<string, Market>
+}
+
+export class GlvTokenTransport extends SubscriptionTransport<GlvTokenTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<GlvTokenTransportTypes>
+  requester!: Requester
+  provider!: ethers.providers.JsonRpcProvider
+  readerContract!: ethers.Contract
+  glvReaderContract!: ethers.Contract
+  abiEncoder!: utils.AbiCoder
+  settings!: GlvTokenTransportTypes['Settings']
+
+  tokensMap: Record<string, Token> = {}
+  marketsMap: Record<string, Market> = {}
+  decimals: Record<string, number> = {}
+
+  async initialize(
+    dependencies: TransportDependencies<GlvTokenTransportTypes>,
+    adapterSettings: GlvTokenTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.settings = adapterSettings
+    this.provider = new ethers.providers.JsonRpcProvider(
+      adapterSettings.ARBITRUM_RPC_URL,
+      adapterSettings.ARBITRUM_CHAIN_ID,
+    )
+    this.readerContract = new ethers.Contract(
+      adapterSettings.READER_CONTRACT_ADDRESS,
+      abi,
+      this.provider,
+    )
+    this.requester = dependencies.requester
+
+    this.glvReaderContract = new ethers.Contract(
+      adapterSettings.GLV_READER_CONTRACT_ADDRESS,
+      glvAbi,
+      this.provider,
+    )
+
+    await this.tokenInfo()
+    await this.marketInfo()
+
+    // setInterval(() => {
+    //   this.tokenInfo()
+    //   this.marketInfo()
+    // }, 60 * 60 * 3 * 1000) // 3 hours
```

Comment:
can this comment block be removed?

Reply:
we need this. but its is interfering with the integration test. 
Waiting for Karen to get back on this issue.